rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -43,31 +43,56 @@ local function get_current_lb_alg()
   return backend["load-balance"] or DEFAULT_LB_ALG
 end
 
+local function is_sticky(backend)
+  return backend["sessionAffinityConfig"] and backend["sessionAffinityConfig"]["name"] == "cookie"
+end
+
 local function balance()
   local backend = get_current_backend()
   if not backend then
     return nil, nil
   end
-  local lb_alg = get_current_lb_alg()
-  local is_sticky = sticky.is_sticky(backend)
-
-  if is_sticky then
-    local endpoint = sticky.get_endpoint(backend)
-    if endpoint ~= nil then
-      return endpoint.address, endpoint.port
-    end
-    lb_alg = "round_robin"
-  end
 
-  if backend["upstream-hash-by"] then
+  if is_sticky(backend) or backend["upstream-hash-by"] then
```

Comment:
It seems like the only reason this `or ` is there is to not repeat fetching the endpoint (L57 - L60). 

You will have some repeated code here - but I would separate the conditions (`is_sticky(backend)` and `backend["upstream-hash-by"]`) to keep it easy to read. This way you could also take out the condition on L62.